### PR TITLE
TEST_EXHAUSTIVE cmake flag & CUDA narrowing warnings

### PR DESCRIPTION
**Pull request by mdavis36**
# Summary

- This PR adds TEST_EXHAUSTIVE as an available flag for CMake.
- Squashes some narrowing conversion warnings when building with CUDA and TEST_EXHAUSTIVE in listsegment-view and rangesegment-view tests

### CODE REVIEW

File: include/RAJA/config.hpp.in [hunk 1]
```diff
@@ -77,6 +77,15 @@
...
 #cmakedefine RAJA_ENABLE_BOUNDS_CHECK
 
+/*
+ ******************************************************************************
+ *
+ * \brief Exhaustive index types for tests
+ *
+ ******************************************************************************
+ */
+#cmakedefine TEST_EXHAUSTIVE
```
Comment (davidbeckingsale) [inline include/RAJA/config.hpp.in:87]: I think we should call it `RAJA_TEST_EXHAUSTIVE`, just in case.

**Comment by trws** [comment]
That makes sense, but then rename all instances of TEST_EXHAUSTIVE we already have to match.

File: include/RAJA/config.hpp.in (hunk 1 above)
Comment (rhornung67) [reply include/RAJA/config.hpp.in]: Yup. @davidbeckingsale beat me to that comment.  BTW, there is a JIRA issue to add 'RAJA_' prefix to all CMake options that are specific to RAJA and cull out any overlaps we have with current BLT or CMake support. I will raise this at tomorrow's meeting. Seems like a good time to get this done.

**Comment by rhornung67** [comment]
@mdavis36 Yup. What @trws just said.

**Pull request merged**